pairing/bn256/gfp.h

```diff
@@ -0,0 +1,32 @@
+#define storeBlock(a0,a1,a2,a3, r) \
```

Comment:
Will this be compiled into the code? That might be worth a comment...

Reply:
cgo will be somewhere if these are to be used, right? I don't see cgo anywhere in this PR

Reply:
I think it's already by default that you can compile this with `go` directly, cgo is included in the default Golang shipment: https://golang.org/cmd/cgo/

Reply:
This is a macro, it is expanded in the .s file.

And none of this is cgo, there is no cgo overhead. It is assembly that the go compiler uses instead of whatever it would've generated for a pure-Go implementation of the same function. Functions are defined in the .s file.